pvlib/ivtools.py

```diff
@@ -262,6 +268,173 @@ def fit_sde_sandia(voltage, current, v_oc=None, i_sc=None, v_mp_i_mp=None,
+    temp_ref : numeric, default 25
+        Reference temperature condition [C]
+
+
+    Returns
+    -------
+    Dictionnary with the following elements:
+
+    * 'I_L_ref': numeric
+        Light-generated current in amperes at std conditions.
+
+    * 'I_o_ref': numeric
+        Diode saturation curent in amperes at std conditions
+
+    * 'R_s': numeric
+        Series resistance in ohms. Note that '_ref' is not mentionned
+        in the name because this resistance is not sensible to the
+        conditions of test.
+
+    * 'R_sh_ref': numeric
+        Shunt resistance in ohms at std conditions.
+
+    * 'a_ref' : numeric
+        Modified ideality factor at std conditions.
+        The product of the usual diode ideality factor (n, unitless),
+        number of cells in series (Ns), and cell thermal voltage at
+        specified effective irradiance and cell temperature.
+
+    * 'alpha_sc': numeric
+        Caution!: Different from the input because of the unit.
+        The short-circuit current (i_sc) temperature coefficient of the
+        module in units of A/K.
+
+    References
+    ----------
+    [1] W. De Soto et al., "Improvement and validation of a model for
+    photovoltaic array performance", Solar Energy, vol 80, pp. 78-88,
+    2006.
+
+    [2] John A Dufﬁe ,William A Beckman, "Solar Engineering of Thermal
+    Processes", Wiley, 2013
+    """
+    # Constants
+    k = 1.381e-23  # Boltzmann constant, in J/K, or 8.617e-5 eV/K
+    q = 1.602e-19  # electron charge in J/V, or 1 eV
+
+    Tref = temp_ref + 273.15  # in K
+
+    if celltype.lower() in ['monosi', 'polysi', 'multisi']:
+        C = 0.0002677  # valid for silicon
+        Eg = 1.796e-19  # in J, valid for silicon
+    elif celltype.lower() in ['gaas']:
+        C = 0.0003174  # valid for gallium arsenide
+        Eg = 2.163e-19  # in J, valid for gallium arsenide
+    elif celltype.lower() in ['cis', 'cigs', 'cdte', 'amorphous']:
+        raise NotImplementedError
+    else:
+        raise ValueError('Unknown cell type.')
+
+    # Conversion from %/K to A/K & V/K
+    alpha_sc = alpha_sc*i_sc/100
+    beta_voc = beta_voc*v_oc/100
+
+    def pv_fct(params, specs):
+        """Returns the system of equations used for computing the
+        single-diode 5 parameters.
+        To avoid the confusion in names with variables of container
+        function the '_' of the variables were removed.
+        """
+        # six input known variables
+        Isc = specs[0]
+        Voc = specs[1]
+        Imp = specs[2]
+        Vmp = specs[3]
+        betaoc = specs[4]
+        alphasc = specs[5]
+        # five parameters vector to find
+        IL = params[0]
+        Io = params[1]
+        a = params[2]
+        Rsh = params[3]
+        Rs = params[4]
+        # five equation vector
+        y = [0, 0, 0, 0, 0]
+
+        # 1st equation - short-circuit - eq(3) in [1]
+        y[0] = Isc - IL + Io*(np.exp(Isc*Rs/a) - 1.0) + Isc*Rs/Rsh
```

Comment:
Consider using [`numpy.expm1`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.expm1.html) here and throughout.

Reply:
Thank you for the feedback.
Is numpy.expm1 (and numpy.logp1 in the same way) prefered for computing efficiency reasons ? because I feel it makes the equation less readable for the standard user who is not an expert in numpy

Reply:
Rationale [here](https://stackoverflow.com/questions/47725508/when-to-use-python-function-expm1-rather-than-exp-1) . Here,  `np.expm1` is preferred when `np.exp(Isc*Rs/a)` is very close to 1 (`Isc*Rs/a` is nearly zero), which is rarely of interest when predicting output using a single diode model, but may be important in this optimization that is attempting to determine the parameters.